### PR TITLE
Too much corn oil can give you a heart attack

### PR DESCRIPTION
**Pull request by yackemflam**
Quit frying/eating so much fried foods you slobs

0 - 15u = no damage, just flavor text
15 - 100u = a slow burn, you probably wont even die from this
100u+ = You WILL get a heart attack, can't get this by solely eating

:cl:
* rscadd: Too much corn oil can damage the heart, watch what you eat

### CODE REVIEW

**Comment by yackemflam** [comment]
travis plz

File: code/modules/reagents/Chemistry-Reagents.dm [hunk 1]
```diff
@@ -4405,6 +4405,33 @@
...
+					holder.remove_reagent(src.id, 0.1 * FOOD_METABOLISM)
+			if(15 to 200)
+				if(prob(10))
+					to_chat(H,"<span class='warning'>You really don't feel very good.</span>")
+				if(prob(5))
+					to_chat(H,"<span class='warning'>You feel a burn in your chest.</span>")
+					var/datum/organ/internal/heart/L = H.internal_organs_by_name["heart"]
+					if(istype(L))
+						L.take_damage(0.2, 1)
+			if(100 to INFINITY)//Too much corn oil holy shit, no one should ever get this high
```
Comment (ghost) [inline code/modules/reagents/Chemistry-Reagents.dm:4425]: dm switch statements break after the first matching condition so this won't actually be reached until 200

File: code/modules/reagents/Chemistry-Reagents.dm [hunk 2]
```diff
@@ -4405,6 +4405,33 @@
...
+					if(H.get_heart())//Got a heart?
+				var/datum/organ/internal/heart/damagedheart = H.get_heart()
+				if (heartdamage >= 30)
+					if(H.species.name != "Diona" && damagedheart) //fuck dionae
```
Comment (ghost) [inline code/modules/reagents/Chemistry-Reagents.dm]: why do you need to check if they're a diona? they shouldn't have a heart anyways and if they do for some reason it won't break anything

**Comment by ArchSunder** [comment]
the scots will remember this

**Comment by Dacendeth** [comment]
Seems kind of weird there's code to specifically fuck diona even though they dont eat.

**Comment by Kharkov72** [comment]
it fucks everything BUT diona for that exact reason

`if(H.species.name != "Diona"`

File: code/modules/reagents/Chemistry-Reagents.dm [hunk 3]
```diff
@@ -4405,6 +4405,33 @@
...
+			if(15 to 200)
+				if(prob(10))
+					to_chat(H,"<span class='warning'>You really don't feel very good.</span>")
+				if(prob(5))
+					to_chat(H,"<span class='warning'>You feel a burn in your chest.</span>")
+					var/datum/organ/internal/heart/L = H.internal_organs_by_name["heart"]
+					if(istype(L))
+						L.take_damage(0.2, 1)
+			if(100 to INFINITY)//Too much corn oil holy shit, no one should ever get this high
+					if(H.get_heart())//Got a heart?
```
Comment (ShiftyRail) [inline code/modules/reagents/Chemistry-Reagents.dm]: Indentation gore

**Comment by Accipitermaris** [comment]
My crystal ball predicts poison-creep

File: code/modules/reagents/Chemistry-Reagents.dm [hunk 4]
```diff
@@ -4405,6 +4405,33 @@
...
 	M.nutrition += nutriment_factor
 
+	if(ishuman(M))
+		var/mob/living/carbon/human/H = M
+		switch(volume)
+			if(1 to 15)
+				if(prob(5))
+					to_chat(H,"<span class='warning'>You burp.</span>")
+					holder.remove_reagent(src.id, 0.1 * FOOD_METABOLISM)
+			if(15 to 200)
```
Comment (Zth--) [inline code/modules/reagents/Chemistry-Reagents.dm]: I think you meant 100 not 200, a few lines below you do `if(100 to INFINITY)`

File: code/modules/reagents/Chemistry-Reagents.dm (hunk 3 above)
Comment (Zth--) [reply code/modules/reagents/Chemistry-Reagents.dm]: what shify said. Also you don't want to do `H.get_heart()` twice

**Review by Zth-- — CHANGES_REQUESTED** [review]
after fixing what's been mentioned please test your code, you didn't even try to compile it

File: code/modules/reagents/Chemistry-Reagents.dm (hunk 4 above)
Comment (yackemflam) [reply code/modules/reagents/Chemistry-Reagents.dm]: It won't really change anything but if your ocd wants it

File: code/modules/reagents/Chemistry-Reagents.dm [hunk 5]
```diff
@@ -4405,6 +4405,33 @@
 
 	M.nutrition += nutriment_factor
 
+	if(ishuman(M))
+		var/mob/living/carbon/human/H = M
+		switch(volume)
+			if(1 to 15)
+				if(prob(5))
+					to_chat(H,"<span class='warning'>You burp.</span>")
```
Comment (Kammerjunk) [inline code/modules/reagents/Chemistry-Reagents.dm]: in space, no-one can hear you burp

**Comment by ghost** [comment]
whoops I reviewed this PR when it went up but apparently you have to hit the submit review button to actually display the comments

File: code/modules/reagents/Chemistry-Reagents.dm (hunk 1 above)
Comment (yackemflam) [reply code/modules/reagents/Chemistry-Reagents.dm:4425]: Already fixed

File: code/modules/reagents/Chemistry-Reagents.dm (hunk 2 above)
Comment (yackemflam) [reply code/modules/reagents/Chemistry-Reagents.dm]: Diona do indeed have hearts; it's a core.
If the core breaks they can't be defibbed

Comment (ghost) [reply code/modules/reagents/Chemistry-Reagents.dm]: my bad, I'm gonna assume the core is a heart subtype so this makes sense

**Comment by yackemflam** [comment]
fixed

**Comment by yackemflam** [comment]
@Zth-- Get to reviewin' brah

File: code/modules/reagents/Chemistry-Reagents.dm (hunk 5 above)
Comment (Zth--) [reply code/modules/reagents/Chemistry-Reagents.dm]: make this a `visible_message` or `emote`

File: code/modules/reagents/Chemistry-Reagents.dm [hunk 6]
```diff
@@ -4397,6 +4397,42 @@
...
+				if(prob(10))
+					to_chat(H,"<span class='warning'>You really don't feel very good.</span>")
+				if(prob(5))
+					to_chat(H,"<span class='warning'>You feel a burn in your chest.</span>")
+					var/datum/organ/internal/heart/L = H.internal_organs_by_name["heart"]
+					if(istype(L))
+						L.take_damage(0.2, 1)
+						
+			if(100 to INFINITY)//Too much corn oil holy shit, no one should ever get this high
+				if(H.get_heart())//Got a heart?
```
Comment (Zth--) [inline code/modules/reagents/Chemistry-Reagents.dm:4426]: this statement is not doing nothing because of the indentation of the next line

**Comment by yackemflam** [comment]
@Zth-- fixed

**Pull request merged**